drivers/serial/stm32-uart.c

```diff
@@ -0,0 +1,91 @@
+#include <sys/types.h>
+
+#include <kernel/cbuf.h>
+#include <kernel/fs.h>
+#include <kernel/irq.h>
+#include <kernel/serial.h>
+#include <kernel/types.h>
+
+#include <cmsis.h>
+#include "platform.h"
+
+static struct cbuf_info cbuf;
+static char buf[16];
+
+static int stm32_getc(struct serial_info *serial, char *c)
+{
+    serial->rx_count--;
+    cbuf_getc(&cbuf, c);
+
+    return 0;
+}
+
+static int stm32_putc(struct serial_info *serial, char c)
+{
+    USART_TypeDef *uart = serial->priv;
+
+    while (!(uart->SR & USART_SR_TXE))
+        ;
+    uart->DR = c;
+
+    return 0;
+}
+
+static int stm32_puts(struct serial_info *serial,
+                      size_t len,
+                      size_t *retlen,
+                      const char *buf)
+{
+    *retlen = len;
+    for (int i = 0; len > 0; len--, i++)
+        stm32_putc(serial, buf[i]);
+
+    return 0;
```

Comment:
The return value is always zero. Is it correct and compliant with Linux fro the perspective of design?

Reply:
I found a similar function, [`static void stm32_transmit_chars(struct uart_port *port)`](https://elixir.free-electrons.com/linux/latest/source/drivers/tty/serial/stm32-usart.c#L279), on Linux v4.13.7.

The design of serial driver on Piko is quite different from one on Linux. 
* On Linux, trig `ops->start_tx(port)` to start up UART to transmit data from local buffers. The transmission flow is invoked in `stm32_transmit_char(port)`.
* On Piko, just trig `ops->stm32_puts(serial, len, retlen, buf)` for transmission.

However, I thought the type of `stm32_puts` return could be `void`.

Reply:
Let's change return type of `stm32_puts` to `void`. You should mention the difference between Piko and Linux within commit messages.